src/merkle/index.rs

```diff
@@ -106,6 +106,13 @@ impl NodeIndex {
         self.value >>= 1;
         self
     }
+
+    /// Traverse one level towards the root, decrementing the depth by `n`.
+    pub fn move_up_by(&mut self, n: u8) -> &mut Self {
+        self.depth = self.depth.saturating_sub(n);
+        self.value >>= n;
+        self
+    }
```

Comment:
Is this used for anything in this PR? If not, maybe skip it for now?

Reply:
Indeed; removed!